### PR TITLE
Use last token and don't try and dereference an empty list

### PR DESCRIPTION
**Pull request by LiamMaru**
Errors were introduced into `ECSService.registerTemplate(...)` during https://github.com/jenkinsci/amazon-ecs-plugin/commit/d2bc8168db56fe606dea5f4cf96bf7ee2b0ad6fb. These errors manifest in two circumstances:

1. There are no task definitions for a given family, or
2. there are more than 100 task definitions for a given family.

In the first case task creation fails because the system tries to retrieve an element from an empty queue. In the second case the system never exits a while-loop and runs out of memory.

This PR resolves both issues. In the former case this PR only tries to retrieve the last element from the queue if the queue is non-empty. In the latter case, the result-set tokens are used to retrieve successive subsets of the complete result-set.

### CODE REVIEW

**Comment by tdonia** [comment]
thanks, @LiamMaru!  This PR fixed the `Slave {0} - Cannot create ECS Task` after having upgraded from 1.5 for us!

**Comment by LiamMaru** [comment]
you're welcome @tdonia, glad I could be of assistance :).

**Comment by dipthegeezer** [comment]
+1 for this I'm getting the same error had to revert to older version of plugin.

File: src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSService.java [hunk 1]
```diff
@@ -194,26 +194,33 @@ String registerTemplate(final ECSCloud cloud, final ECSTaskTemplate template, St
...
+        boolean templateMatchesExistingContainerDefinition = false;
+        boolean templateMatchesExistingVolumes = false;
+        boolean templateMatchesExistingTaskRole = false;
 
-        boolean templateMatchesExistingTaskRole = template.getTaskrole() == null || template.getTaskrole().equals(describeTaskDefinition.getTaskDefinition().getTaskRoleArn());
-        LOGGER.log(Level.INFO, "Match on task role: {0}", new Object[] {templateMatchesExistingTaskRole});
-        LOGGER.log(Level.FINE, "Match on task role: {0}; template={1}; last={2}", new Object[] {templateMatchesExistingTaskRole, template.getTaskrole(), describeTaskDefinition.getTaskDefinition().getTaskRoleArn()});
+        DescribeTaskDefinitionResult describeTaskDefinition = null;
 
+        if(taskDefinitions.size() > 0) {
```
Comment (roehrijn) [inline src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSService.java]: .isEmpty() would be better here, but I'm going to merge anyway

**Comment by LiamMaru** [comment]
No worries @roehrijn. Re. The size() > 0, I agree... Unfortunately I wrote and submitted this fix at around 3:00 AM! We'll get it right eventually :-).

**Pull request merged**